### ISSUE TITLE
[Feature Request] Settings presets/profiles

### ISSUE THREAD

**Issue by murk-sy** [prusa3d/Prusa-CW-Firmware#38]
This is more an elaboration on #37 than a completely new feature request.

I propose adding options of up to 5 (or at the very least 3) presets which can be easily switched between. Extremely useful if you're working with multiple different resins.

The menu structure would be as follows:

> - *Start drying/curing (without tank)*
> - *Start washing (only with tank inserted)*
> - *Resin preheat*
> - Profiles
>   - Profile 1: W3 D5 C5
>   - Profile 2: W10 D30 C30
>   - Profile 3: W3x2 C3
>   - Save to profile
>     - Save to profile 1
>     - Save to profile 2
>     - Save to profile 3
>   - Default
> - *Run-Time* (should be moved under Settings)
> - *Settings*

Profile selection should show the basic settings: `W`ash time (and cycles?), `D`ry time and `C`ure time and display them depending on run mode.
Setting profile names would be more work than it's worth anyway, but having those settings available at a glance would make it very easy to discern profiles to some degree.
Default would load factory settings that can also be saved to profiles. This isn't necessary, but it would be nice to have available.
Another not necessary but nice feature would be current active profile number displayed by just a number in the corner of the screen  (unless modified).

Regardless of the displayed, each profile should save the following settings:
- Run mode
- Curing run-time
- Drying run-time
- Washing run-time
- Wash cycles (if implemented)
- Resin preheat time
- Curing speed
- Washing speed
- Warm-up - On / Off
- Drying warm-up temperature
- Resin preheat temperature

Although with that, it would be recommended to separate profile settings from machine settings (like LCD brightness, fan speed, etc), perhaps by marking them with an asterisk.

**Comment by tritol** [comment]
Yes, you've described the features I've been thinking about for over a year. Unfortunately, the company's priorities are elsewhere and I would like to do different things in my free time.

**Comment by murk-sy** [comment]
Understandable, and I'd prefer developers don't spend their free time looking at work either.

If it's priorities, that's perfectly fine - there's quite a few products in the pipeline, and this is just a QOL feature. I just hope SL1S/CW1S will have a set spot in the line, rather than being delayed for other, newer priorities.

Hopefully this FR be useful as sort of a baseline when the time comes. I'll edit it if I get any interesting related to that.
